client/state/comments/actions.js

```diff
@@ -0,0 +1,303 @@
+ * @returns {Function} a thunk that creates a comment for a given post
+ */
+export function writeComment( commentText, siteId, postId, parentCommentId ) {
+	if ( ! commentText || ! siteId || ! postId ) {
+		return;
+	}
+
+	return ( dispatch ) => {
+		const placeholderComment = createPlaceholderComment( commentText, postId, parentCommentId );
+
+		// Insert a placeholder
+		dispatch( {
+			type: COMMENTS_RECEIVE,
+			siteId,
+			postId,
+			comments: [ placeholderComment ]
+		} );
+
+		let apiPromise;
+
+		if ( parentCommentId ) {
+			apiPromise = wpcom.site( siteId ).post( postId ).comment( parentCommentId ).reply( {}, commentText );
+		} else {
+			apiPromise = wpcom.site( siteId ).post( postId ).comment().add( commentText );
+		}
+
+		return apiPromise.then( ( comment ) => {
+			// remove the placeholder
+			dispatch( {
+				type: COMMENTS_REMOVE,
+				siteId,
+				postId,
+				commentId: placeholderComment.ID
+			} );
+
+			// insert the real comment
+			dispatch( {
+				type: COMMENTS_RECEIVE,
+				siteId,
+				postId,
+				comments: [ comment ]
+			} );
+
+			return comment;
+		} )
+		.catch( ( error ) => {
+			dispatch( {
+				type: COMMENTS_ERROR,
+				siteId,
+				postId,
+				commentId: placeholderComment.ID,
+				error
+			} );
+
+			throw error;
+		} );
+	};
+}
+
+/***
+ * Creates a thunk that likes a comment
+ * @param {Number} siteId site identifier
+ * @param {Number} postId post identifier
+ * @param {Number} commentId comment identifier
+ * @returns {Function} think that likes a comment
+ */
+export function likeComment( siteId, postId, commentId ) {
+	return ( dispatch ) => {
+		// optimistic update
+		dispatch( {
+			type: COMMENTS_LIKE,
+			siteId,
+			postId,
+			commentId
+		} );
+
+		// optimistic revert on error, return here for test more conveniently
+		return wpcom.site( siteId ).comment( commentId ).like().add( { source: 'reader' } ).then( ( data ) => dispatch( {
+			type: COMMENTS_LIKE_UPDATE,
+			siteId,
+			postId,
+			commentId,
+			iLike: data.i_like,
+			likeCount: data.like_count
+		} ) ).catch( () => dispatch( {
+			type: COMMENTS_UNLIKE,
+			siteId,
+			postId,
+			commentId
+		} ) );
+	};
+}
+
+/***
+ * Creates a thunk that unlikes a comment
+ * @param {Number} siteId site identifier
+ * @param {Number} postId post identifier
+ * @param {Number} commentId comment identifier
+ * @returns {Function} think that unlikes a comment
+ */
+export function unlikeComment( siteId, postId, commentId ) {
+	return ( dispatch ) => {
+		// optimistic update
+		dispatch( {
+			type: COMMENTS_UNLIKE,
+			siteId,
+			postId,
+			commentId
+		} );
+
+		// optimistic revert on error, return here for test more conveniently
+		return wpcom.site( siteId ).comment( commentId ).like().del( { source: 'reader' } ).then( ( data ) => dispatch( {
+			type: COMMENTS_LIKE_UPDATE,
+			siteId,
+			postId,
+			commentId,
+			iLike: data.i_like,
+			likeCount: data.like_count
+		} ) ).catch( () => dispatch( {
```

Comment:
@gwwar That doesn't fetches, it can be seen in the tests, since `nock` blocks all outgoing requests.

@blowery I'm not sure I'm following. In order for a user to be able to 'like' a comment they need to click on the like button, which rendered on a comment, therefor the comment is in state. 

For the sake of argument let's say he/she somehow succeed in doing so, were we expect to see a result of such like? Since we don't have a comment for it, the current reducer will just ignore it.

I also not sure I'm following your comment regarding the `catch`, currently the way it works is:
- User clicks the like button, that dispatches immidiate +1 on the like count and i_like -> true. 
- Somewhat later, we get a success result from wpcom, that result contains updated likes count (including the like above and maybe some more likes), we set that count for the comment.
- We might also get a failure from wpcom, then we need to remove the +1 we made and i_like -> false.

at least imo, that's what the code does =)